tools/c7n_azure/c7n_azure/resources/sqlserver.py

```diff
@@ -89,9 +97,134 @@ def _query_rules(self, resource):
+
+    schema = type_schema(
+        'set-firewall-rules',
+        rinherit=SetFirewallAction.schema,
+        **{
+            'bypass-rules': {'type': 'array', 'items': {
+                'enum': ['AzureServices']}},
+        }
+    )
+
+    def __init__(self, data, manager=None):
+        super(SqlSetFirewallAction, self).__init__(data, manager)
+        self.log = log
+
+    def _process_resource(self, resource):
+        # Get existing rules
+        old_ip_rules = list(self.client.firewall_rules.list_by_server(
+            resource['resourceGroup'],
+            resource['name']))
+        old_ip_space = [IPRange(r.start_ip_address, r.end_ip_address) for r in old_ip_rules]
+
+        # Build new rules
+        new_ip_rules = self._build_ip_rules(old_ip_space, self.data.get('ip-rules', []))
+
+        # Normalize data types into IPNetwork and IPRange
+        new_ip_space = self._normalize_rules(new_ip_rules)
+
+        # Build bypass rules
+        # SQL uses a 0.0.0.0 rule to track "Azure Services" bypass
+        old_bypass = []
+        if AZURE_SERVICES in old_ip_space:
+            old_bypass.append('AzureServices')
+
+        new_bypass = self.data.get('bypass-rules', old_bypass)
+        if 'AzureServices' in new_bypass and AZURE_SERVICES not in new_ip_space:
+            new_ip_space.append(AZURE_SERVICES)
+
+        # Update ARM resources
+        to_remove_ip_space = set(old_ip_space).difference(new_ip_space)
+        for r in to_remove_ip_space:
+            remove = next(i for i in old_ip_rules
+                          if i.start_ip_address == str(IPAddress(r.first)) and
+                          i.end_ip_address == str(IPAddress(r.last)))
+            self.client.firewall_rules.delete(
+                resource['resourceGroup'],
+                resource['name'],
+                remove.name
+            )
+
+        to_add_ip_space = set(new_ip_space).difference(old_ip_space)
+        for r in to_add_ip_space:
+            first = IPAddress(r.first)
+            last = IPAddress(r.last)
+            self.client.firewall_rules.create_or_update(
+                resource['resourceGroup'],
+                resource['name'],
+                self._generate_rule_name(r),
+                str(first),
+                str(last)
+            )
+
+        return 'Added {} rules, removed {} rules.'.format(
+            len(to_add_ip_space), len(to_remove_ip_space))
+
+    def _normalize_rules(self, new_ip_rules):
+        new_ip_space = []
+        for rule in new_ip_rules:
+            if '-' in rule:
+                parts = rule.split('-')
+                new_ip_space.append(IPRange(parts[0], parts[1]))
+            else:
+                net = IPNetwork(rule)
+                new_ip_space.append(IPRange(net.first, net.last))
+        return new_ip_space
+
+    def _generate_rule_name(self, rule):
+        if rule == AZURE_SERVICES:
+            return 'AllowAllWindowsAzureIps'
+        return "c7n-" + str(uuid.uuid4())
```

Comment:
can we make it configurable? (As for network rules)

Reply:
Oh darn, yes I meant for it to be and completely forgot!